File: scripts/matterLib.js
function commonItem(Name, color, explosivenessI, flammabilityI, radioactivityI, chargeI, hardnessI, unlock) {
    var i = extend(Item, Name, color, {
        explosiveness: explosivenessI,
        flammability: flammabilityI,
        radioactivity: radioactivityI,
        charge: chargeI,
        hardness: hardnessI,
        alwaysUnlocked: unlock,
    });
    if (unlock !== true) { unlock = false } else { unlock = true };
    return i = i;
    exports.i = i;
};
exports.commonItem = commonItem;

File: src/WithoutCore/libs/MatterLib.java
package WithoutCore.libs;

import mindustry.type.Item;
import arc.graphics.Color;

//interface staticItem(){};
public class MatterLib {
    public static void staticItem(String name, Color color, boolean unlock, boolean canbuild, float expAbility,
            float flaAbility,
            float radAbility, float chaAbility, int ness) {
        new Item(name, color) {
            {
                alwaysUnlocked = unlock;
                buildable = canbuild;
                explosiveness = expAbility;
                flammability = flaAbility;
                radioactivity = radAbility;
                charge = chaAbility;
                hardness = ness;
            }
        };
    };

    public void dynamicItem() {
    };
}
